test/pipe.js

```diff
@@ -50,23 +50,54 @@ assert.equal(shell.error(), null);
 assert.equal(result.toString(), shell.cat('resources/uniq/pipeSorted').toString());
 
 // Synchronous exec
-// TODO: add windows tests
-if (process.platform !== 'win32') {
+if (process.platform === 'win32') {
+  // Windows-specific
+
+  // Ensure the user does not have the wrong version of FIND. I.e.,
+  // require the following to return error.
+  result = shell.exec('CD..&FIND . -name no');
+  if (!shell.error()) {
+    console.error('Warning: Cannot verify piped exec: Found POSIX-like find(1) in PATH. This test assumes a Windows environment with its FIND command (fix your PATH, exit cygwin/mingw32/MSYS2).');
+  } else {
+    // “CD..” is to avoid Windows running test/find.js with Windows
+    // Script Host. I tried using regex to remove ‘.’ from PATH but
+    // apparently exec sources the OS’s environment when executing which
+    // ends up re-adding ‘.’ to PATH.
+    result = shell.cat('resources/grep/file').exec('CD..&FIND "alph"');
+    assert.ok(!shell.error());
+    assert.equal(result, 'alphaaaaaaabeta\r\nalphbeta\r\n');
+  }
+} else {
   // unix-specific
   if (shell.which('grep').stdout) {
     result = shell.cat('resources/grep/file').exec("grep 'alpha*beta'");
-    assert.equal(shell.error(), null);
+    assert.ok(!shell.error());
     assert.equal(result, 'alphaaaaaaabeta\nalphbeta\n');
   } else {
     console.error('Warning: Cannot verify piped exec');
   }
-} else {
-  console.error('Warning: Cannot verify piped exec');
 }
 
 // Async exec
-// TODO: add windows tests
-if (process.platform !== 'win32') {
+if (process.platform === 'win32') {
+  // Windows-specified
```

Comment:
nit: you can delete this comment, it's implied by the conditional. Same thing for the section above.